src/harness/unittests/languageService.ts

```diff
@@ -0,0 +1,49 @@
+/// <reference path="..\harness.ts" />
+
+namespace ts {
+    describe("languageService", () => {
+        const files: {[index: string]: string} = {
+            "foo.ts": `import Vue from "./vue";
+import Component from "./vue-class-component";
+import { vueTemplateHtml } from "./variables";
+
+@Component({
+    template: vueTemplateHtml,
+})
+class Carousel<T> extends Vue {
+}`,
+            "variables.ts": `export const vueTemplateHtml = \`<div></div>\`;`,
+            "vue.d.ts": `export namespace Vue { export type Config = { template: string }; }`,
+            "vue-class-component.d.ts": `import Vue from "./vue";
+export function Component(x: Config): any;`
+};
+        it("should be able to create a language service which can respond to deinition requests without throwing", () => {

```

Comment:
It's not clear from reading this test that this has anything to do with stringwriter -- could you use a smaller repro?

Reply:
Not really, no. The repro occurs in trying to print the symbol for the reference for go to definition, which requires the module symbol the containing type is from be aliased cross-file for the repro to occur. Like, it probably doesn't have to explicitly be a decorator and a class in the root file; but it wouldn't cut out much making it into a call expression.

Reply:
The problem is that the above paragraph isn't clear from a test that just says `should be able to create a language service which can respond to deinition requests without throwing` -- we already have lots of tests for definition requests that don't throw! Could you add a comment at least?

Reply:
Will do